xinference/model/rerank/core.py

```diff
@@ -265,7 +265,12 @@ def rerank(
         if max_chunks_per_doc is not None:
             raise ValueError("rerank hasn't support `max_chunks_per_doc` parameter.")
         logger.info("Rerank with kwargs: %s, model: %s", kwargs, self._model)
-        sentence_combinations = [[query, doc] for doc in documents]
+        from .utils import pre_instruction
+
+        instruction = pre_instruction(
```

Comment:
Maybe just create a `preprocess_sentence`, here we need to concatenate the query and sentence, maybe it can be other ways, all the things could be included into the function.